File: civet/death-rolls.js
let numberOfPlayers = 0

class Player {
  name = `Player ${++numberOfPlayers}`;
  lastRoll = 1000
  roll(startingNumber) {
    let ret;
    ret = this.lastRoll = Math.floor((Math.random() * startingNumber) + 1)
    console.log(`Random! ${this.name} rolls a ${this.lastRoll} (out of ${startingNumber}).`)
    return ret
  }
};

let [currentPlayer, otherPlayer] = [new Player, new Player]

while (!(otherPlayer.lastRoll === 1)) {
  currentPlayer.roll(otherPlayer.lastRoll);
  [currentPlayer, otherPlayer] = [otherPlayer, currentPlayer]
}

console.log(`Game over! ${currentPlayer.name} wins!`)

File: dart/death-rolls/death-rolls.dart
import "dart:math";

class Player {
  static int numberOfPlayers = 0;
  late String name;
  int lastRoll = 1000;

  Player() {
    numberOfPlayers++;
    name = "Player $numberOfPlayers";
  }

  roll(int startingNumber) {
    lastRoll = Random().nextInt(startingNumber) + 1;
    print("Random! $name rolls a $lastRoll (out of $startingNumber).");
  }
}

main() {
  var currentPlayer = Player();
  var otherPlayer = Player();

  while (otherPlayer.lastRoll != 1) {
    currentPlayer.roll(otherPlayer.lastRoll);
    var temp = currentPlayer;
    currentPlayer = otherPlayer;
    otherPlayer = temp;
  }

  print("Game over! ${currentPlayer.name} wins!");
}
